OpenRA.Mods.Cnc/Traits/Disguise.cs

```diff
@@ -188,10 +188,13 @@ public void DisguiseAs(Actor target)
 				}
 				else
 				{
-					var tooltip = target.TraitsImplementing<ITooltip>().FirstOrDefault();
-					AsPlayer = tooltip.Owner;
-					AsActor = target.Info;
-					AsTooltipInfo = tooltip.TooltipInfo;
+					var tooltip = target.TraitsImplementing<ITooltip>().FirstEnabledTraitOrDefault();
+					if (tooltip != null)
```

Comment:
This change would leave `AsPlayer`, `AsActor`, and `AsTooltipInfo` unchanged, which is the wrong thing for this method to do.

I deliberately left this one out of my list of simple fixes because this is a design problem that will need a larger rework, not just a simple code oversight.

Reply:
I see. Removed it again.